src/operations/query/filters/reference.js

```diff
@@ -34,13 +34,14 @@ function filterByReference(propertyObj, and_segments, queryParameterValue, colum
             );
         }
     } else {
+        var field = propertyObj.fields ? `${propertyObj.fields[1]}` : `${propertyObj.field}.reference`;
```

Comment:
The first index accessed here is at 'fields[1]', do we want to check the length of the fields first? or is there code up top that checks for this already?

Reply:
it will only have 'field_s_' (vs field) if there is more than one value in the array. So I felt it was safe to lean on it.

Reply:
I think thats correct most of the time.  It's just that 'fields' on the backend is just an array object, so it could only be a size of one on user input error.

Reply:
yeah okay. I'll fix it, thanks